crates/providers/lattice-controller/src/lib.rs

```diff
@@ -0,0 +1,630 @@
+        // so there is nothing to clean up per link removal
+    }
+
+    /// Handle shutdown request
+    #[instrument(level = "debug", skip(self))]
+    async fn shutdown(&self) {
+        // No cleanup necessary for shutdown, since links do not
+        // determine/regulate the NATS connections
+    }
+}
+
+/// Implement the lattice-controller-provider provider contract specified in WIT (see provider.wit)
+#[async_trait]
+impl WasmcloudLatticeControlLatticeController for LatticeControllerProvider {
+    /// Sets lattice credentials and stores them in the cache to be used to create a
+    /// connection in the next operation that requires one
+    #[instrument(level = "debug", skip(self, _ctx, arg), fields(actor_id = ?_ctx.actor, lattice_id = ?arg.lattice_id))]
+    async fn set_lattice_credentials(
+        &self,
+        _ctx: Context,
+        arg: SetLatticeCredentialsRequest,
+    ) -> ProviderInvocationResult<CtlOperationAck> {
+        let config = ConnectionConfig {
+            cluster_uris: vec![arg
+                .nats_url
+                .clone()
+                .unwrap_or_else(|| "0.0.0.0:4222".to_string())],
+            auth_jwt: arg.user_jwt.clone(),
+            auth_seed: arg.user_seed.clone(),
+            lattice_prefix: arg.lattice_id.to_string(),
+            js_domain: arg.js_domain.clone(),
+            ..Default::default()
+        };
+
+        // Setting the auction_timeout to a large value may trigger timeouts in distant code.
+        //
+        // - host configuration rpc_timeout
+        // - `LatticeControllerSender`s which have embedded RPC clients w/ internal timeouts
+        // - httpserver/messaging provider(s) which expect a certain response speed from calling actors
+        //
+        // Since auctions will *wait* until the auction_timeout to do operations like gathering hosts,
+        // we must manually ensure this value is unlikely to cause timeouts.
+        let host_data = wasmcloud_provider_sdk::load_host_data().map_err(|e| {
+            ProviderInvocationError::Provider(format!("failed to load host data: {e}"))
+        })?;
+        if host_data
+            .default_rpc_timeout_ms
+            .is_some_and(|v| v < config.timeout_ms)
+        {
+            warn!(
+                host_rpc_timeout_ms = host_data.default_rpc_timeout_ms,
+                auction_timeout = config.auction_timeout_ms,
+                "host default RPC timeout < auction timeout, operations that rely on auctions are likely to time out"
+            );
+        }
+
+        self.get_connections()
+            .await
+            .put_config(&arg.lattice_id, config)
+            .await;
+
+        Ok(CtlOperationAck {
+            accepted: true,
+            error: "".to_string(),
+        })
+    }
+
+    /// Sets registry credentials, storing them in the cache
+    #[instrument(level = "debug", skip(self, _ctx, arg), fields(actor_id = ?_ctx.actor, lattice_id = ?arg.lattice_id))]
```

Comment:
Nit: change to `skip_all`

Reply:
- [ ] Fix `skip_all` for `instrument` call